.github/workflows/soak.yml

```diff
@@ -0,0 +1,244 @@
+          file: soaks/Dockerfile
+          builder: ${{ steps.buildx.outputs.name }}
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha, scope=${{ github.workflow }}
+          cache-to: type=gha, scope=${{ github.workflow }}
+          build-args: |
+            VECTOR_FEATURES=api,api-client,disk-buffer,enrichment-tables,rdkafka-plain,sinks,sources,sources-dnstap,transforms,unix,vendor-all,vrl-cli,datadog-pipelines
+
+  build-comparison-image:
+    name: Build comparison 'soak-vector' container
+    runs-on: [self-hosted, linux, x64, general]
+    steps:
+      - uses: actions/checkout@v2.3.5
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          flavor: |
+            latest=false
+            prefix=
+            suffix=
+          images: ghcr.io/${{ github.repository }}/soak-vector
+          tags: type=raw, value=${{ github.sha }}
+
+      - name: Build and push 'soak-vector' image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          file: soaks/Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha, scope=${{ github.workflow }}
+          cache-to: type=gha, scope=${{ github.workflow }}
+          build-args: |
+            VECTOR_FEATURES=api,api-client,disk-buffer,enrichment-tables,rdkafka-plain,sinks,sources,sources-dnstap,transforms,unix,vendor-all,vrl-cli,datadog-pipelines
+
+  soak-baseline:
+    name: Soak (${{ matrix.target }}) - baseline
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    runs-on: [self-hosted, linux, x64, soak]
+    needs: [build-baseline-image]
+    strategy:
+      matrix:
+        target:
+          - datadog_agent_remap_datadog_logs
+          - syslog_humio_logs
+          - syslog_log2metric_humio_metrics
+          - syslog_loki
+          - syslog_regex_logs2metric_ddmetrics
+          - syslog_splunk_hec_logs
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Download baseline-soak artifact
+        uses: actions/download-artifact@v1
+        with:
+          name: baseline-soak
+
+      - name: Run baseline experiment
+        run: |
+          mkdir -p /tmp/${{ matrix.target }}/
+          ./soaks/bin/soak_one.sh "false" ${{ matrix.target }} "baseline" `cat baseline-soak/sha` /tmp/${{ matrix.target }}
```

Comment:
The runners aren't cleaned in-between runs. I assume that's fine? Or should this do an `rm` before just to be safe?

Reply:
TIL

I will clean the runners.

Reply:
Resolved in 77f5ee991c6346818cd99d65b9137b646c1af63c